### PR TITLE
Task 05: User factory

### PR DESCRIPTION
**Pull request by KaplenkoAnastasiia**

### CODE REVIEW

File: tasks/task05/task05.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def create_user(name, user_name,  age = 42, **keywords):
```
Comment (ahodkov) [inline tasks/task05/task05.py]: `**kwargs`

File: tasks/task05/task05.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  extra = {}
+  for kw in keywords:
```
Comment (ahodkov) [inline tasks/task05/task05.py]: `**kwargs` is dict and you don't need to unpack.

A good tutorial for this: [here](https://habr.com/ru/company/ruvds/blog/482464/)

File: tasks/task05/task05.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    extra[kw] = keywords[kw]
+  return print({
```
Comment (ahodkov) [inline tasks/task05/task05.py]: You have not fully understood the meaning of the factory. Factory pattern needed to create an object. In your case, you can't use objects in another place.
A good tutorial for all patterns is: [here](https://refactoring.guru/ru/design-patterns/creational-patterns)

File: tasks/task05/task05.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    "name": name,
+    "surname": user_name,
+    "age": age,
+    "extra": extra
+  })
+
+create_user("John", "Doe")
```
Comment (ahodkov) [inline tasks/task05/task05.py]: `print(create_user("John", "Doe"))`

File: tasks/task05/task05.py (hunk 1 above)
Comment (ahodkov) [reply tasks/task05/task05.py]: A good point is when you define an object at the function level.
Check it out: `def create_user(name: str, user_name:str, age:int = 42, **keywords):`
In your case, the user can define age to string or boolean

File: tasks/task05/task05.py (hunk 3 above)
Comment (ahodkov) [reply tasks/task05/task05.py]: I mean: You need to return object

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: tasks/task05/task05.py (hunk 1 above)
Comment (KaplenkoAnastasiia) [reply tasks/task05/task05.py]: thanks

File: tasks/task05/task05.py (hunk 3 above)
Comment (KaplenkoAnastasiia) [reply tasks/task05/task05.py]: changed

File: tasks/task05/task05.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+def create_user(name:str, user_name:str,  age:int = 42, **extra):
```
Comment (ahodkov) [inline tasks/task05/task05.py]: `create_user(first_name: str, last_name: str, age:int = 42, **kwargs):`
This is much better to understand and `**kwargs` is better not to rename.

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: tasks/task05/task05.py (hunk 5 above)
Comment (KaplenkoAnastasiia) [reply tasks/task05/task05.py]: ok

**Review by ahodkov — APPROVED** [review]

**Pull request merged**